CMakeLists.txt

```diff
@@ -29,18 +29,27 @@ endif()
 if(NOT BOOST_SUPERPROJECT_VERSION)
     option(BOOST_URL_INSTALL "Install boost::url files" ON)
     option(BOOST_URL_BUILD_TESTS "Build boost::url tests" ${BUILD_TESTING})
+    option(BOOST_URL_BUILD_EXAMPLES "Build boost::url examples" ${BOOST_URL_IS_ROOT})
 else()
     set(BOOST_URL_BUILD_TESTS ${BUILD_TESTING})
 endif()
 
 
 include(GNUInstallDirs)
 if(BOOST_URL_IS_ROOT)
+    if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/../../CMakeLists.txt")
+        set(BOOST_URL_FIND_PACKAGE_BOOST ON)
+    endif()
+
     # container comes from json
     set(BOOST_INCLUDE_LIBRARIES url json)
     set(BOOST_EXCLUDE_LIBRARIES url)
     set(CMAKE_FOLDER _deps)
-    add_subdirectory(../.. _deps/boost EXCLUDE_FROM_ALL)
+    if (BOOST_URL_FIND_PACKAGE_BOOST)
+        find_package(Boost 1.78.0 REQUIRED)
```

Comment:
This makes it possible to test the library locally with containers. I can keep this local if you want.